app/models/workflow.rb

```diff
@@ -6,6 +6,29 @@ class Workflow < ApplicationRecord
   validates :display_name, presence: true
 
   def groups
-    transcriptions.group(:group_id).count
+    # aggregate group_id, transcription count, and last update info in one db call
```

Comment:
I'm trying to unpack this but not getting anywhere. I think the `worklfow.group` method name here (which is very close to `Workflow.group`) is not very descriptive. What is the intention of this 'group' query lookup? 

It looks like the nested sub-select query is grouping the `workflow.translations.group(:group_id)` but also counts the transcription data and selects the most recently updated_at (should this use a limit 1?)

It then joins the sub-select on max_date to find more transcriptions and combine the counts...

sadly i don't follow. A nice description of what is needed here vs just the 'group' wording would help.  

Also can the Join please be explicit (INNER JOIN) vs a default?

FWIW - i'd add comments to this method / query to explain what the statements are trying to achieve. 

Also did anyone run an explain on this query to make sure it was hitting the db indexes / it's a scalable query once data starts filling up the transcriptions / workflows tables?

Reply:
@camallen , thanks for bringing this up. It's really great to know when code that I've written is hard to read/follow. Lets hash this out over a call when you're back from holiday.